src/TelnetClient.php

```diff
@@ -238,6 +253,15 @@ protected function getResponse($prompt = null, $promptError = null)
                 $isError = true;
                 break;
             }
+
+            // throw an exception if the number of characters read is greater than the limit
+            if ($this->maxReadCharacters > 0 && strlen($buffer) >= $this->maxReadCharacters) {
+                throw new TelnetException(sprintf(
+                    'Maximum number of characters read (%d), the last characters were %s',
```

Comment:
Printing the last 10 bytes read is good idea but might look a little wonky if you are in the middle of multibyte characters. Having said that, I think it's fine for now.

Reply:
Cool, hopefully this will give more visibility about what is happening.

Reply:
Yes, absolutely. Better to see _something_ than nothing.